flight3handler.go

```diff
@@ -193,7 +193,17 @@ func handleServerKeyExchange(_ flightConn, state *State, cfg *handshakeConfig, h
 			return &alert.Alert{Level: alert.Fatal, Description: alert.InternalError}, err
 		}
 		state.IdentityHint = h.IdentityHint
-		state.preMasterSecret = prf.PSKPreMasterSecret(psk)
+		if state.cipherSuite == nil || !state.cipherSuite.UsesEllipticCurves() {
```

Comment:
I think `state.cipherSuite == nil` should be fatal. Maybe return a `alert.Fatal`?

It is great to check, but undefined behavior if we fall into the else I am worried about.

Reply:
fixed